core/src/test/scala/integration/kafka/api/BaseConsumerTest.scala

```diff
@@ -274,9 +279,24 @@ abstract class BaseConsumerTest extends IntegrationTestHarness with Logging {
   }
 
   protected def sendRecords(numRecords: Int, tp: TopicPartition) {
-    (0 until numRecords).map { i =>
-      this.producers(0).send(new ProducerRecord(tp.topic(), tp.partition(), s"key $i".getBytes, s"value $i".getBytes))
-    }.foreach(_.get)
+    sendRecords(this.producers(0), numRecords, tp)
+    this.producers(0).flush()
+  }
+
+  protected def sendCompressedRecords(numRecords: Int, tp: TopicPartition) {
```

Comment:
Can we make this `sendRecords` and have it take a compression-codec argument instead of hard-coding to gzip?